gateway/src/apicast/oauth/oidc.lua

```diff
@@ -30,20 +31,34 @@ local mt = {
 
 local empty = {}
 
-function _M.new(oidc_config)
+function _M.new(oidc_config, service)
   local oidc = oidc_config or empty
   local issuer = oidc.issuer
   local config = oidc.config or empty
   local alg_values = config.id_token_signing_alg_values_supported or empty
 
+  local tmpl, tmpl_err, tmpl_type
   local err
+
+  if oidc.client_id then
+    tmpl_type = oidc.client_id_type
+    tmpl, tmpl_err = TemplateString.new(oidc.client_id, tmpl_type)
+    if tmpl_err then
+      err = 'Invalid client_id template string'
+    end
+
+  end
+
   if not issuer or #alg_values == 0 then
     err = 'missing OIDC configuration'
   end
 
   return setmetatable({
     config = config,
     issuer = issuer,
+    service = service,
+    tmpl = tmpl,
```

Comment:
I think that having more descriptive names in these attributes would make the code easier to understand.